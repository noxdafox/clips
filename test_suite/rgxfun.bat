(clear)
(regex-match)
(regex-match foo 1)
(regex-match foo "1")
(regex-match foo bar)
(regex-match foo foobar)
(regex-match foo "foobar")
(regex-match "foo" foobar)
(regex-match "foo" "foobar")
(regex-match "" foobar)
(regex-match foo FOOBAR)
(regex-match ☃ "Foo☃Bar")
(regex-match "(www.)(.*)(.com|.net)" "http://www.clipsrules.net")
(regex-imatch)
(regex-imatch foo 1)
(regex-imatch foo "1")
(regex-imatch foo bar)
(regex-imatch foo foobar)
(regex-imatch foo "foobar")
(regex-imatch "foo" foobar)
(regex-imatch "foo" "foobar")
(regex-imatch "" foobar)
(regex-imatch foo FOOBAR)
(regex-imatch ☃ "Foo☃Bar")
(regex-imatch "(www.)(.*)(.com|.net)" "http://www.clipsrules.net")
