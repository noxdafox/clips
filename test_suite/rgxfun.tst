(unwatch all)
(clear)
(set-strategy depth)
(open "Results//rgxfun.rsl" rgxfun "w")
(dribble-on "Actual//rgxfun.out")
(batch "rgxfun.bat")
(dribble-off)
(load "compline.clp")
(printout rgxfun "rgxfun.bat differences are as follows:" crlf)
(compare-files "Expected//rgxfun.out" "Actual//rgxfun.out" rgxfun)
; close result file
(close rgxfun)
