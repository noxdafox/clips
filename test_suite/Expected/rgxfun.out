TRUE
CLIPS> (clear)
CLIPS> (regex-match)
[ARGACCES4] Function regex-match expected exactly 2 argument(s)
CLIPS> (regex-match foo 1)
[ARGACCES5] Function regex-match expected argument #2 to be of type symbol, string, or instance name
CLIPS> (regex-match foo "1")
()
CLIPS> (regex-match foo bar)
()
CLIPS> (regex-match foo foobar)
("foo")
CLIPS> (regex-match foo "foobar")
("foo")
CLIPS> (regex-match "foo" foobar)
("foo")
CLIPS> (regex-match "foo" "foobar")
("foo")
CLIPS> (regex-match "" foobar)
("")
CLIPS> (regex-match foo FOOBAR)
()
CLIPS> (regex-match ☃ "Foo☃Bar")
("☃")
CLIPS> (regex-match "(www.)(.*)(.com|.net)" "http://www.clipsrules.net")
("www.clipsrules.net" "www." "clipsrules" ".net")
CLIPS> (regex-imatch)
[ARGACCES4] Function regex-imatch expected exactly 2 argument(s)
CLIPS> (regex-imatch foo 1)
[ARGACCES5] Function regex-imatch expected argument #2 to be of type symbol, string, or instance name
CLIPS> (regex-imatch foo "1")
()
CLIPS> (regex-imatch foo bar)
()
CLIPS> (regex-imatch foo foobar)
("foo")
CLIPS> (regex-imatch foo "foobar")
("foo")
CLIPS> (regex-imatch "foo" foobar)
("foo")
CLIPS> (regex-imatch "foo" "foobar")
("foo")
CLIPS> (regex-imatch "" foobar)
("")
CLIPS> (regex-imatch foo FOOBAR)
("FOO")
CLIPS> (regex-imatch ☃ "Foo☃Bar")
("☃")
CLIPS> (regex-imatch "(www.)(.*)(.com|.net)" "http://www.clipsrules.net")
("www.clipsrules.net" "www." "clipsrules" ".net")
